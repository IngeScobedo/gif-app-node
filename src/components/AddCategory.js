import { useState } from "react";
import PropTypes from "prop-types";
import "./AddCategory.css";

const AddCategory = ({ setCategories }) => {
  const [inputValue, setInputValue] = useState("Search...");
  let handleInputChange = (e) => {
    let nValue = e.target.value;
    setInputValue(nValue);
  };
  const handleSubmit = (e) => {
    e.preventDefault();
    console.log('PreventDefault', inputValue);
    if(inputValue.length > 2){
      setCategories((category) => [inputValue, ...category]);
      resetInput();
    }
  };
  const resetInput = () => {
    setInputValue("");
  };
  return (
    <form onSubmit={handleSubmit}>
      <p>{inputValue}</p>
      <input
        type="text"
        value={inputValue}
        onChange={handleInputChange}
        onClick={resetInput}
      />
    </form>
  );
};
AddCategory.propTypes = {
  setCategories: PropTypes.func.isRequired,
};
export default AddCategory;
