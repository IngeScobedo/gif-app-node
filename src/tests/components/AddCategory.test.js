import {shallow} from 'enzyme'
import AddCategory from '../../components/AddCategory'


describe('Testing <AddCategory/>', ()=>{

    let setCategories = jest.fn();

    const value = 'hello world';

    let wrapper = shallow(<AddCategory setCategories={setCategories} />)

    beforeEach(() => {
        jest.clearAllMocks();
        wrapper = shallow(<AddCategory setCategories={setCategories} />)

    })
    test('Debe mostrarse correctamente', () => {
        expect(wrapper).toMatchSnapshot();
    })

    test('Debe cambiar la caja de texto', () => {
        let input = wrapper.find('input');
        input.simulate('click');
        input.simulate( 'change',{ target: {value} });
        expect(wrapper.find('p').text()).toBe(value);
    })
    test('No debe de poster la caja de texto', () => {
        let input = wrapper.find('input');
        input.simulate('click');
        wrapper.find('form').simulate('submit', {preventDefault(){ }} );
        expect(setCategories).not.toHaveBeenCalled();
    })
    
    test('Debe de llamar al setCategories y limpiar la caja de texto', () => {
        //1.- Simular inputChange
        let input = wrapper.find('input');
        input.simulate('click');
        input.simulate( 'change',{ target: {value} });
        expect(wrapper.find('p').text()).toBe(value);
        //2.- Simular submit
        wrapper.find('form').simulate('submit', {preventDefault(){ }} );        
        //3.- SetCategories se debe haber llamado
        expect(setCategories).toHaveBeenCalledTimes(1);
        //4.- El valor del input debe estar vacio
        expect(input.text()).toBe('')
    })
    
} )

