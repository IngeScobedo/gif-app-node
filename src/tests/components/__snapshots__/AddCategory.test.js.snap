// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing <AddCategory/> Debe mostrarse correctamente 1`] = `
<form
  onSubmit={[Function]}
>
  <p>
    Search...
  </p>
  <input
    onChange={[Function]}
    onClick={[Function]}
    type="text"
    value="Search..."
  />
</form>
`;
